scene/3d/spatial.cpp

```diff
@@ -188,7 +188,9 @@ void Spatial::_notification(int p_what) {
 					if (data.gizmo.is_valid()) {
 						data.gizmo->create();
 						if (data.gizmo->can_draw()) {
-							data.gizmo->redraw();
+							if (is_visible_in_tree()) {
```

Comment:
Maybe we can save one indentation level and just do `if (is_visible_in_tree() && data.gizmo->can_draw())`?

Reply:
Yes, that was my first implementation. But after taht I saw  Spatial::_update_gizmo() and changed it to keep consistency.

Reply:
Ok, fair enough :)